pkg/subscriber/github/subscriber_item.go

```diff
@@ -950,33 +959,46 @@ func (ghsi *SubscriberItem) sortResources(repoRoot string, resourcePath string)
 	return nil
 }
 
-func (ghsi *SubscriberItem) sortKubeResources(path string) error {
+func (ghsi *SubscriberItem) sortKubeResource(path string) error {
 	if strings.EqualFold(filepath.Ext(path), ".yml") || strings.EqualFold(filepath.Ext(path), ".yaml") {
 		klog.V(4).Info("Reading file: ", path)
 
-		file, _ := ioutil.ReadFile(path)
-		t := kubeResource{}
-		err := yaml.Unmarshal(file, &t)
+		file, err := ioutil.ReadFile(path) // #nosec G304
 
 		if err != nil {
-			fmt.Println("Failed to unmarshal YAML file")
+			klog.Error(err, "Failed to read YAML file "+path)
 			return err
 		}
 
-		if t.APIVersion != "" && t.Kind != "" {
-			if strings.EqualFold(t.Kind, "customresourcedefinition") {
-				ghsi.crdsAndNamespaceFiles = append(ghsi.crdsAndNamespaceFiles, path)
-			} else if strings.EqualFold(t.Kind, "namespace") {
-				ghsi.crdsAndNamespaceFiles = append(ghsi.crdsAndNamespaceFiles, path)
-			} else if strings.EqualFold(t.Kind, "serviceaccount") {
-				ghsi.rbacFiles = append(ghsi.rbacFiles, path)
-			} else if strings.EqualFold(t.Kind, "clusterrole") {
-				ghsi.rbacFiles = append(ghsi.rbacFiles, path)
-			} else if strings.EqualFold(t.Kind, "role") {
-				ghsi.rbacFiles = append(ghsi.rbacFiles, path)
-			} else {
-				ghsi.otherFiles = append(ghsi.otherFiles, path)
+		resources := utils.ParseKubeResoures(file)
+
+		if len(resources) == 1 {
+			t := kubeResource{}
+			err := yaml.Unmarshal(resources[0], &t)
+
+			if err != nil {
+				fmt.Println("Failed to unmarshal YAML file")
+				// Just ignore the YAML
+				return nil
+			}
+
+			if t.APIVersion != "" && t.Kind != "" {
```

Comment:
Hi Roke, 

I'm missing some context over here, why do we want to handle these kinds separately?

Reply:
The input to our system is a YAML with multiple kube resources. The output of our system that the kube client API we use takes one kube resource object to apply. So within our system, we need to handle this type of YAML, separate all resources and feed one resource at a time to the kube client.

Reply:
Yes, I understand that. 

Maybe my question is not clear, I meat to ask, in this `if` block, we are sperate `ghsi.rbacFiles`, `ghsi.crdsAndNamespaceFiles` against`ghsi.otherFiles `. I mean, why do you need this kind separation.

Reply:
When there are mixture of kube resource kinds, we need to ensure that certain kinds of resources are applied first. For example, without a namespace, applying resouces into that namespace will fail. Without RBAC resource, applying other resources depending on the RBAC will fail. Line 150 applies these in such order. This ordering might not be complete today. Using kustomize ensures that the resource kinds are applied in correct order.

Reply:
Nice, new knowledge gained, thanks.

I was thinking, we will just let the resource fail, and the second or later sync loop will pick it up and apply.